django/db/models/sql/compiler.py

```diff
@@ -978,12 +982,14 @@ def _get_field_choices():
             if klass_info is None:
                 invalid_names.append(name)
                 continue
-            select_index = klass_info['select_fields'][0]
-            col = self.select[select_index][0]
-            if self.connection.features.select_for_update_of_column:
-                result.append(self.compile(col)[0])
-            else:
-                result.append(self.quote_name_unless_alias(col.alias))
+            select_fields = klass_info['select_fields']
+            if select_fields:
```

Comment:
IMO this is not a proper solution because it simple ignores all models, except the  queryset's model.